server/utils/checkYourAnswersUtils.ts

```diff
@@ -310,7 +310,8 @@ export const removeAnyOldPageKeys = (questions: any, task: string, applicationPa
   const latestPageKeys = Object.keys(questions[task])
   const matchedKeys = applicationPageKeys.filter(
```

Comment:
should we rename this function, do you think? It seems to be more about removing page keys that don't contain data, rather than removing old page keys

Reply:
Yeah, I think this logic could do with a refactor, but it's out of scope of this work.